driver/addlayernorm_driver.hpp

```diff
@@ -0,0 +1,531 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#ifndef GUARD_MIOPEN_ADDLAYERNORM_DRIVER_HPP
+#define GUARD_MIOPEN_ADDLAYERNORM_DRIVER_HPP
+
+#include <../test/tensor_holder.hpp>
+#include <../test/verify.hpp>
+#include "InputFlags.hpp"
+#include "driver.hpp"
+#include "random.hpp"
+#include "tensor_driver.hpp"
+#include "timer.hpp"
+#include <algorithm>
+#include <cfloat>
+#include <cstdlib>
+#include <memory>
+#include <miopen/tensor.hpp>
+#include <numeric>
+#include <vector>
+
+template <typename Tgpu, typename Tcheck>
+int32_t mloAddLayerNormForwardRunHost(miopenTensorDescriptor_t inputDesc,
+                                      Tgpu* input,
+                                      Tgpu* input2,
+                                      Tgpu* weight,
+                                      Tgpu* bias,
+                                      Tcheck* outputhost,
+                                      Tcheck* meanhost,
+                                      Tcheck* rstdhost,
+                                      float eps,
+                                      int32_t normalized_dim,
+                                      miopenNormMode_t mode)
+{
+    auto dims         = miopen::deref(inputDesc).GetLengths();
+    size_t outer_size = 1;
```

Comment:
It wouldn't hurt to have a sanity check on normalized_dim here. The valid range isn't clear to me yet, but something like:
```suggestion
    MIOPEN_THROW_IF(normalized_dim < 0 || static_cast<size_t>(normalized_dim) >= dims.size(), "normalized_dim out of range");
    size_t outer_size = 1;
```

Reply:
fixed it.